### ISSUE TITLE
Flickering white bar on the left of the source

### ISSUE THREAD

**Issue by Mindstormer619** [mltframework/shotcut#418]
## Specs

- OS: Linux Ubuntu 16.04
- Shotcut: 17.04.02

## Overview

There is a persistent flickering white bar on the leftmost edge of added sources. This is not just a viewer glitch -- the exported video file also contains the same flickering bar.

The source in question was recorded using [OBS Studio](https://obsproject.com/). The original source file has no such flickering. It's pretty much a Shotcut-induced thing.

## Screen capture

 https://www.useloom.com/share/189346d7adeb4bfd94eea161323eb1a2

**Comment by ddennedy** [comment]
What is the resolution of the video clip, and what is the Shotcut Settings > Video Mode set to?

**Comment by Mindstormer619** [comment]
Resolution: 1364x768. 
Video mode is set to Automatic.

![image](https://cloud.githubusercontent.com/assets/441849/25063505/6726c9f4-2203-11e7-8210-68d6facc221b.png)

My screen res is 1366x768. Is it the two missing pixels in the width that's causing it or something?

**Comment by ddennedy** [comment]
I believe I reproduced this, and it is regression introduced in Shotcut v17.03. Can you test if the problem occurs in v17.02 for you to confirm this? You can get it from [here](https://github.com/mltframework/shotcut/releases/tag/v17.02) and you do not have to uninstall current version to use it.

**Comment by Mindstormer619** [comment]
Can confirm, the problem **does not occur** on Shotcut 17.02.05.

**Comment by visionshen** [comment]
hi Dan, so what is the root cause

**Comment by ddennedy** [comment]
This caused by videos with certain widths and the sliced pixel format conversion we started using since a couple of Shotcut versions for performance improvement. The sliced pixel conversion is using FFmpeg library swscale, and some optimizations in it may be overwriting neighboring bytes and pixels when the slice width is not some multiple of 8. With a video such as you if I set environment variable MLT_AVFORMAT_SLICED_PIXFMT_DISABLE=1 while running Shotcut, it turns off the sliced conversion, and the problem goes away. I probably need to come up with a heuristic in the MLT avformat module such that if the last slice is not some multiple of 8, 16, or 32; then, we not use slices.
